Gp/release management scripts

JackPiri opened the pull request:
This PR targets:

- release preparation script
  - added configurability (skip/stop) for initialization "check main checked out" and "check no pending changes" steps, in order to allow more flexibility in case of non-standard usage of script
  - added step for checking for dependencies availability
  - remove `authors.md` file update (removal of file itself)
  - added update of `CLIENT_VERSION_IS_RELEASE` macro
 
- release backport script
  - added script for release backporting (from the docs: _"A successful run of the script would create a new branch starting from currently selected commit of `main` branch (a different branch can be configured) and would perform one single commit containing all the release backport steps."_)

- update/removal of obsolete files (scripts and docs)

JackPiri committed:
Release preparation Python script update

Removed authors.md file update
Made configurable the constraint of running the script with main branch checked out
Add check for depends packages availability
Add update of macro CLIENT_VERSION_IS_RELEASE
Skipping a script step has priority over stopping-on

JackPiri committed:
Added release backport Python script

Additional minor refactoring of release preparation script
Updated/removed obsolete files (scripts and docs)

a-petrini commented on doc/release-process.md:
```diff
@@ -1,156 +1,45 @@
+    - at least 100 confirmations
+	- timestamps of 3 preceding and following blocks are all in series
+	- no forked blocks in 3 preceding and following blocks (check via getchaintips RPC command on running node)
+	  and "total_transactions" are set based
```
Is this line incomplete?

JackPiri replied:
You are right, _and "total_transactions" are set based_ is a leftover; took the chance also for fixing indentation.

a-petrini commented on doc/release-process.md:
```diff
@@ -1,156 +1,45 @@
+- push release preparation local branch to remote branch
+- create associated release branch on remote (note this release branch is different from release preparation branch)
+- open a PR merging release preparation into release branch and wait for approval
+- create annotated tag (vX.Y.Z)
```
I would add that tag creation and push should be performed after the release preparation branch has been merged into the release branch

JackPiri replied:
Ok makes sense, modified as per your suggestion.

a-petrini commented on contrib/devtools/release_management/release_preparation.py:
```diff
@@ -0,0 +1,465 @@
+    if (len(sys.argv) > 1):
+        config_file = sys.argv[1]
+    else:
+        # config_file = "config.yaml"
```
Can we remove this comment?

JackPiri replied:
For sure, removed comment from both `release_preparation.py` and `release_backport.py`.

a-petrini replied:
Thanks, everything looks fine and I resolve all conversations

a-petrini reviewed (approved):
PR looks fine! Left a few non blocking suggestions

drgora reviewed (approved):
Looks good to me, thanks

ptagl commented on doc/release-process.md:
```diff
@@ -1,156 +1,44 @@
+- push release preparation local branch to remote branch
+- create associated release branch on remote (note this release branch is different from release preparation branch)
+- open a PR merging release preparation into release branch and wait for approval
+- once the release preparation branch is merged into the release branch, create annotated tag (vX.Y.Z)
```
Just as a reminder, in particular for me as I always forget that detail, can we mention the full command to create the signed annotated tag?

Not sure if this applies also to `contrib/devtools/README.md` as I see some overlapping.

```suggestion
- once the release preparation branch is merged into the release branch, create annotated tag (vX.Y.Z), for example `git tag -s -m"v4.0.0" v4.0.0`
```

JackPiri replied:
I've modified a little bit differently as `git tag -a v4.0.0 -m "v4.0.0"` (please note `-a` instead of `-s`).

JackPiri replied:
Sorry, finally modified with your proposal.

ptagl commented on README.md:
```diff
@@ -1,4 +1,4 @@
-Zen 4.0.99
+Zend 4.0.99
```
Even though the main readme file has always included the current release version, it looks weird to me to see it now as X.Y.99 as it is not a real release.

What about getting rid of the version? In case, we would need to update the release script to skip this file.

```suggestion
Zend
```

JackPiri replied:
Ok, updated single file and scripts step accordingly.

JackPiri committed:
General refactoring for release management Python scripts

Folders tree reorganization and duplicated code removal
Release notes file ("-current.md") proper cleaning
Removed README.md file update/reset
